clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -272,35 +267,23 @@ public boolean handlesResourceType(ResourceType<?> resourceType) {
   @Override
   public PersistenceSpaceIdentifier getPersistenceSpaceIdentifier(String name, CacheConfiguration<?, ?> config) throws CachePersistenceException {
     DefaultClusterCacheIdentifier identifier = new DefaultClusterCacheIdentifier(name);
-    Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository> existing = knownPersistenceSpaces
-        .putIfAbsent(name, new Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository>(identifier, new ClusteredMapRepository()));
-    if (existing != null) {
-      identifier = existing.first;
-    }
     return identifier;
   }
 
-  private ConcurrentHashMap<EntityRef<ConcurrentClusteredMap, Object>, ConcurrentClusteredMap<?, ?>> getNewMapEntityMap() {
-    return new ConcurrentHashMap<EntityRef<ConcurrentClusteredMap, Object>, ConcurrentClusteredMap<?, ?>>();
-  }
-
   @Override
   public void releasePersistenceSpaceIdentifier(PersistenceSpaceIdentifier<?> identifier) throws CachePersistenceException {
-    if (!isKnownIdentifier(identifier)) {
+    ClusteredCacheIdentifier clusterCacheIdentifier = (ClusteredCacheIdentifier) identifier;
+    if (!knownPersistenceSpaces.containsKey(clusterCacheIdentifier)) {
```

Comment:
This can be replaced by the remove call directly:
`if (knownPersistenceSpaces.remove(clusterCacheIdentifier) == null)`